### PR TITLE
DAOS-11151 build: Update to pmdk 1.12.0

### PR DESCRIPTION
**Pull request by jolivier23**
In preparation for 1.12.1 which promises to fix a major bug

PR-repos: pmdk@PR-23

Signed-off-by: Jeff Olivier <jeffrey.v.olivier@intel.com>

Required-githooks: true
Signed-off-by: Jeff Olivier <jeffrey.v.olivier@intel.com>

### CODE REVIEW

**Comment by jolivier23** [comment]
EL8 VM tests failure is https://daosio.atlassian.net/browse/DAOS-10383

File: utils/scripts/install-centos7.sh [hunk 1]
```diff
@@ -57,3 +57,4 @@ dnf --nodocs install \
     sudo \
     valgrind-devel \
     yasm
+ln -s /usr/bin/cmake3 /usr/bin/cmake
```
Comment (jolivier23) [inline utils/scripts/install-centos7.sh:60]: Simplest workaround for a platform we are discontinuing.

**Review by soumagne — DISMISSED** [review]

File: utils/scripts/install-centos7.sh (hunk 1 above)
Comment (brianjmurrell) [reply utils/scripts/install-centos7.sh:60]: I don't think that will do what you want.  That is altering the environment that `mock` (the tool that builds the chroot that the RPM build is done in) is running in, not the environment that `rpmbuild` runs in.  There is no way to achieve your simple hack above in that chroot.

File: utils/rpms/daos.spec [hunk 2]
```diff
@@ -166,12 +166,10 @@ Requires: ndctl
...
 %if (0%{?suse_version} >= 1500)
 Requires: ipmctl >= 02.00.00.3733
-# When 1.11.2 is released, we can change this to >= 1.11.2
-Requires: libpmemobj1 = 1.11.0-3.suse1500
+Requires: libpmemobj1 >= 1.12.0
 %else
 Requires: ipmctl > 02.00.00.3816
-# When 1.11.2 is released, we can change this to >= 1.11.2
-Requires: libpmemobj = 1.11.0-3%{?dist}
+Requires: libpmemobj >= 1.12.0
```
Comment (brianjmurrell) [inline utils/rpms/daos.spec:172]: Ditto.

**Comment by brianjmurrell** [comment]
> Test stage Functional on EL 8 completed with status FAILURE. https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-9779/11/display/redirect

There was a `.toLowerCase()` that needed dropping in the `Jenkinsfile` that seems to have not gotten applied in this PR.

File: utils/rpms/daos.spec [hunk 3]
```diff
@@ -166,12 +166,10 @@ Requires: ndctl
 # needed to set PMem configuration goals in BIOS through control-plane
 %if (0%{?suse_version} >= 1500)
 Requires: ipmctl >= 02.00.00.3733
-# When 1.11.2 is released, we can change this to >= 1.11.2
-Requires: libpmemobj1 = 1.11.0-3.suse1500
+Requires: libpmemobj1 >= 1.12.0
```
[the remark this answers is not in the document]
Comment (jolivier23) [reply utils/rpms/daos.spec:169]: I think when they release the fix in 1.12.1, we definitely do not want to use a prior version...but for here, maybe not

File: utils/scripts/install-centos7.sh (hunk 1 above)
Comment (brianjmurrell) [reply utils/scripts/install-centos7.sh:60]: More to the point, how did you even figure out you needed this if we are not building on CentOS 7 any more?

**Review by brianjmurrell — APPROVED** [review]

File: utils/scripts/install-centos7.sh (hunk 1 above)
Comment (jolivier23) [reply utils/scripts/install-centos7.sh:60]: Because we still run unit tests on Centos 7

**Pull request closed without merging**